packages/databyss-api/src/models/Source.js

```diff
@@ -14,21 +14,24 @@ const SourceSchema = new mongoose.Schema({
   },
   text: {
     type: String,
-    required: true,
+    // required: true,
   },
-  ranges: { type: Array },
-  citations: {
+  name: {
     type: String,
+    //  required: true,
   },
+  ranges: { type: Array },
```

Comment:
Ranges applies to a single citation, i.e. each citation in `citations` has a `ranges` field.
I think we can use mongoose [subdocuments](https://mongoosejs.com/docs/subdocs.html) for this. So we'd have a `richText` schema: `const RichTextSchema = new Schema({ textValue: { type: String }, ranges: { type: Array })` and then in the source schema: `citations: [RichTextSchema]`.

We should do the same with `ranges`, as in declare a `RangeSchema` that defines the fields of a range.